tests/phpunit/WebTest/Contact/GroupAddTest.php

```diff
@@ -317,4 +317,53 @@ public function testAddSmartGroup() {
     $this->assertEquals($count, trim($expectedCount[1]));
   }
 
+  /**
+   *  group update_smart_groups action test
+   */
+  public function testGroupUpdateSmartGroupsForRelatedContacts() {
```

Comment:
I don't how the other tests would work. Did try to run the test with closest example i could find.

Reply:
So this is probably the test I would start with

https://github.com/davecivicrm/civicrm-core/blob/fae75b583b102cf9cfdc64ba0adc08091f8dd5a8/tests/phpunit/api/v3/ContactTest.php#L3523

It already loads multiple groups - so that is good - but none of them have this specific related contact criteria that is hitting the problem

Reply:
(I'm trying to find a test that creates a smart group now)

Reply:
OK - this is from ActionSchedule Test

```
    // create smart group which will contain all Male contacts
    $smartGroupParams = array('formValues' => array('gender_id' => 1));
    $smartGroupID = $this->smartGroupCreate(
      $smartGroupParams,
      array(
        'name' => 'new_smart_group',
        'title' => 'New Smart Group',
        'parents' => array($groupID => 1),
      )
    );
   // create related group 2
   // create a contact that would be in only 1
   $result = $this->callApiSuccess('Contact', 'get', ['group' => $group1['id']);
    $result2 $this->callApiSuccess('Contact', 'get', ['group' => $group2['id']);
  // Check result is not cross-contaminated
```

So something like 
```
public function testLoadMultipleRelatedGroups
    // create smart group which will contain all Male contacts
    $smartGroupParams = array('formValues' => array(... related group params 1....));
    $smartGroupID = $this->smartGroupCreate(
      $smartGroupParams,
      array(
        'name' => 'new_smart_group',
        'title' => 'New Smart Group',
        'parents' => array($groupID => 1),
      )
    );
```

Reply:
I tried to perform a unit-test. I get an error. Can you direct me to a guide for preparing to run the test.
Here's the command.
```
phpunit4 --filter testAddCreateIndividual tests/phpunit/api/v3/ContactTest.php
```
Error
```
exception 'RuntimeException' with message '_populateDB requires CIVICRM_UF=UnitTests'
```

Reply:
I normally run tests from within phpunit - this is what my command line (generated by phpunit) ooks like

```
phpunit.php --bootstrap /Users/emcnaughton/buildkit/build/dmaster/sites/all/modules/civicrm/tests/phpunit/CiviTest/bootstrap.php --configuration /Users/emcnaughton/buildkit/build/dmaster/sites/all/modules/civicrm/phpunit.xml.dist --filter "/::testAddCreateIndividual( .*)?$/" api_v3_ContactTest /Users/emcnaughton/buildkit/build/dmaster/sites/all/modules/civicrm/tests/phpunit/api/v3/ContactTest.php
```

Reply:
Ohh - looks like this is specifically relevant https://docs.civicrm.org/dev/en/latest/testing/phpunit/